src/Dependencies/Collections/SegmentedDictionary`2.cs

```diff
@@ -0,0 +1,1629 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis.Collections.Internal;
+
+namespace Microsoft.CodeAnalysis.Collections
+{
+    /// <summary>
+    /// Represents a collection of keys and values.
+    /// </summary>
+    /// <remarks>
+    /// <para>This collection has the same performance characteristics as <see cref="Dictionary{TKey, TValue}"/>, but
+    /// uses segmented arrays to avoid allocations in the Large Object Heap.</para>
+    /// </remarks>
+    /// <typeparam name="TKey">The type of the keys in the dictionary.</typeparam>
+    /// <typeparam name="TValue">The type of the values in the dictionary.</typeparam>
+    [DebuggerTypeProxy(typeof(IDictionaryDebugView<,>))]
+    [DebuggerDisplay("Count = {Count}")]
+    internal class SegmentedDictionary<TKey, TValue> : IDictionary<TKey, TValue>, IDictionary, IReadOnlyDictionary<TKey, TValue>
```

Comment:
sealed? 

Don't see much reason to allow derivation in our context.

Reply:
Now sealed